hacking/test-module

```diff
@@ -141,18 +142,43 @@ def boilerplate_module(modfile, args, interpreter, check, destfile):
         task_vars=task_vars
     )
 
+    if module_style == 'new' and 'ZIPLOADER_WRAPPER = True' in module_data:
+        module_style = 'ziploader'
+
     modfile2_path = os.path.expanduser(destfile)
     print("* including generated source, if any, saving to: %s" % modfile2_path)
-    print("* this may offset any line numbers in tracebacks/debuggers!")
+    if module_style not in ('ziploader', 'old'):
+        print("* this may offset any line numbers in tracebacks/debuggers!")
     modfile2 = open(modfile2_path, 'w')
     modfile2.write(module_data)
     modfile2.close()
     modfile = modfile2_path
 
-    return (modfile2_path, module_style)
+    return (modfile2_path, modname, module_style)
+
+def ziploader_setup(modfile, modname):
+    os.system("chmod +x %s" % modfile)
+
+    cmd = subprocess.Popen([modfile, 'explode'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = cmd.communicate()
+    lines = out.splitlines()
+    if len(lines) != 2 or 'Module expanded into' not in lines[0]:
+        print("*" * 35)
+        print("INVALID OUTPUT FROM NEW-STYLE MODULE WRAPPER")
```

Comment:
I personally don't find the term 'new' stands up that well, maybe something like:

`INVALID OUTPUT FROM ZIPLOADED-STYLE MODULE WRAPPER`